src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -921,4 +921,45 @@ class DeltaSuite extends QueryTest
       checkAnswer(spark.table("sc15200test"), Seq.empty)
     }
   }
+
+  test("support for setting dataChange to false") {
+    withTempDir { tempDir =>
+      if (tempDir.exists()) {
+        assert(tempDir.delete())
+      }
+      val deltaLogUrl = DeltaLog.forTable(spark, tempDir).snapshot.path
+
+      spark.range(100)
+        .repartition(4)
+        .write
+        .format("delta")
+        .save(tempDir.toString)
+
+      val df = spark.read.format("delta").load(tempDir.toString)
+
+      df
+        .repartition(2)
+        .write
+        .format("delta")
+        .mode("overwrite")
+        .option("dataChange", "false")
+        .save(tempDir.toString)
+
+      val removesDataChange = spark.read
+        .json(deltaLogUrl + "/00000000000000000001.json")
+        .where("remove IS NOT NULL").select($"remove.dataChange")
+        .as[Boolean]
+        .collect()
+        .toSeq
+
+      val addsDataChange = spark.read
+        .json(deltaLogUrl + "/00000000000000000001.json")
+        .where("add IS NOT NULL").select($"add.dataChange")
+        .as[Boolean]
+        .collect()
+        .toSeq
```

Comment:
As Rahul mentioned, let's be a little smarter on how we get these values. You can use `deltaLog.snapshot.version` to get the current version, `deltaLog.store.read(deltaFile(deltaLog.logPath, version)).map(Action.fromJson)` to get the actions in a commit, and then `val fileActions = commitActions.collect { case a: FileAction => a }` to get the AddFiles and RemoveFiles from that commit.

Reply:
OK, I will do it using DeltaLog.snapshot